internal/pkg/cli/task_run.go

```diff
@@ -377,6 +393,77 @@ func (o *runTaskOpts) Validate() error {
 		}
 	}
 
+	for _, value := range o.secrets {
+		//For Systems Manager Parameter Store parameter you can specify it as name or the ARN if it exists in the same Region as the task you are launching.
+		if !isSSM(value) && !isSecretsManager(value) {
+			return fmt.Errorf("must specify a valid secrets ARN")
+		}
+	}
+
+	return nil
+}
+
+func isSSM(value string) bool {
+	// For SSM parameter you can specify it as ARN or name if it exists in the same Region as the task you are launching.
+	return !template.IsARNFunc(value) || strings.Contains(value, ":ssm:")
+}
+
+func isSecretsManager(value string) bool {
+	return template.IsARNFunc(value) && strings.Contains(value, ":secretsmanager:")
+}
+
+func (o *runTaskOpts) getCategorizedSecrets() (map[string]string, map[string]string) {
+	if len(o.ssmParamSecrets) > 0 || len(o.secretsManagerSecrets) > 0 {
+		return o.secretsManagerSecrets, o.ssmParamSecrets
+	}
+
+	for name, value := range o.secrets {
+		if isSSM(value) {
+			o.ssmParamSecrets[name] = value
+		}
+		if isSecretsManager(value) {
+			o.secretsManagerSecrets[name] = value
+		}
+	}
+	return o.secretsManagerSecrets, o.ssmParamSecrets
+}
+
+func (o *runTaskOpts) confirmSecretsAccess() error {
+	if o.executionRole != "" {
+		return nil
+	}
+
+	if o.appName != "" || o.env != "" {
+		return nil
+	}
+
+	if o.acknowledgeSecretsAccess {
+		return nil
+	}
+
+	secretsManagerSecrets, ssmParamSecrets := o.getCategorizedSecrets()
+
+	log.Infoln("Looks like you're requesting ssm:GetParameters to the following SSM parameters:")
+
+	for _, value := range ssmParamSecrets {
+		log.Infoln("* " + value)
+	}
+
+	log.Infoln("\nand secretsmanager:GetSecretValue to the following Secrets Manager secrets:")
+
+	for _, value := range secretsManagerSecrets {
+		log.Infoln("* " + value)
+	}
```

Comment:
We should consider the case when only either one of ssm or secretsmanager type is specified. You could get a prompt that look like:
```
Looks like you're requesting ssm:GetParameters to the following SSM parameters:
* some secrets

and secretsmanager:GetSecretValue to the following Secrets Manager secrets:

  Do you grant permission to the ECS/Fargate agent for these secrets? [? for help] (y/N)
```
with the current implementation.